logic.js

```diff
@@ -0,0 +1,176 @@
+    for(let i=0; i<array.length; i++){
+        if(array[i].getId() == id){
+            // splice function does not remove the last item in the mutated array, that's why we have to reassign an empty array
+            if(array.length == 1){
+                array = []
+                break
+            }
+            console.log('the element that should be removed is of id: '+array[i].getId())
+            console.log('i: '+i)
+            let deleted = array.splice(i, 1)
+            
+            for(let k=0; k<deleted.length; k++){
+                console.log('spliced item is '+deleted[k].getId())
+            }
+            
+            console.log('Remaining elements:')
+            for (let j=0; j<array.length; j++){
+                console.log(array[j].getId())
+            }
+            break;
+        }
+    }
+    return array
+}
+
+// get element from an array, returns null of not found 
+const getElementById = (array, id) => {
+    for(let i=0; i<array.length; i++){
+        console.log(`tweet id = ${array[i].getId()} ${id}`)
+        if(array[i].getId() == id){
+            return array[id]
+        }
+    }
+    return null
+}
+
+// add element to the end of the array
+const addElement = (array, element) => {
+    array.push(element);
+} 
+
+// Tweets module which is the list of all posted tweets
+// ===================================================================
+const Tweets = function(){
+    // List of all tweets
+    let allTweets = []
+
+    // tweet module
+    /* it is configured inside Tweets module, because a tweet can not exist without Tweets modules
+       which refers to the list of all the tweets*/
+    // ========================================================
+    const Tweet = function(tweetContent){
+        let tweetText = tweetContent
+        let tweetId = 0
+        let comments = []
+        // comment module
+        /*it is configured inside Tweet module, because a cooment can not exist without the post
+        that it is refering to*/
+        // ====================================================
+        const Comment = function(commentContent) {
+            let commentText = commentContent
+            let commentId = 0
+            const getcommentContent = ()=> commentText
+            const setcommentContent = (value) => {
+                commentText= value
+            }
+            const getcommentId = ()=> commentId
+            const setcommentId = (value) => {
+                commentId= value
+            }
+            return{
+                getcommentContent,
+                setcommentContent,
+                getId: getcommentId,
+                setId: setcommentId
+            }
+        }
+        // =====================End of comment module==================================
+        const getTweetContent = () => tweetText
+        const setTweetContent = (value) => {
+            tweetText = value;
+        }
+
+        const getTweetId = () => tweetId
```

Comment:
What's the idea of creating a function to return the id, if you can access the variable himself?